### DISCUSSION TITLE
Why OpenSSL?

### DISCUSSION THREAD

**Discussion by EliahKagan** [GitoxideLabs/gitoxide#1494] [Q&A] [answered]
The `max` build configuration uses OpenSSL instead of rustls, suggesting that there is an advantage of doing so. Is this for performance—does OpenSSL perform better?—to decrease executable size, to use local configuration for OpenSSL, or some other reason?

(I think this is a separate question from #1493, though I acknowledge that I may have a misunderstanding about OpenSSL involved in both. This can be closed if it is better answered there.)

**Comment by Byron** [comment, marked as answer]
That's a good question - in theory, using the `rustls` shoudl always be preferable. In practice, it's probably about OpenSSL being able to find the trust store more reliably - so I vaguely remember there are issues with compatibility with `rustls` and it has its own problems.

`curl` is what pulls all this in, I believe.

**Reply by NobodyXu** [reply]
There's one more motivation to use rustls: it supports http3 (quic), where as openssl hasn't added support for it yet (last I heard they are working on it).

**Reply by EliahKagan** [reply]
The `[lib]curl` aspect of this is being discussed in #1493, so I'll mark this answered. The discussion can of course still be moved back here if that's more convenient.
